From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sat, 4 May 2024 15:43:58 +0800
Subject: [PATCH] UI&ARC: AuxiliaryTools
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

整理AuxiliaryTable
MinRi2 <2275045670@qq.com>

修复ScriptButtons
way-zer <himc.wicp@gmail.com> on 2024/4/27 at 17:08

从ARC拆分，并重命名
way-zer <himc.wicp@gmail.com> in 2024/5/4

修复锁定视角行为错误
way-zer <himc.wicp@gmail.com> on 2024/6/10

从ARC中完全分离
way-zer <himc.wicp@gmail.com> on 2024/6/27

* 调整部分按钮位置
* 合并PC/Mobile的脱离视角功能
* 优化辅助器波次显示UI
* clean
way-zer <himc.wicp@gmail.com> on 2024/9/8
---
 core/src/mindustry/ui/fragments/HudFragment.java | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/src/mindustry/ui/fragments/HudFragment.java b/core/src/mindustry/ui/fragments/HudFragment.java
index 37a57b425a423ddff029268c938423f6711340c8..6a9359568e73967b34be82618ec7a95583e5cf6e 100644
--- a/core/src/mindustry/ui/fragments/HudFragment.java
+++ b/core/src/mindustry/ui/fragments/HudFragment.java
@@ -43,6 +43,7 @@ public class HudFragment{
 
     private ImageButton flip;
     public NewCoreItemsDisplay coreItems = new NewCoreItemsDisplay();
+    private final AuxiliaryTools auxiliaryTools = new AuxiliaryTools();
 
     private String hudText = "";
     private boolean showHudText;
@@ -121,6 +122,9 @@ public class HudFragment{
             .touchable(Touchable.disabled)
             .style(Styles.outlineLabel)
             .name("position");
+            if(Core.settings.getInt("AuxiliaryTable") == 3){
+                t.row().add(auxiliaryTools);
+            }
             t.top().right();
         });
 
@@ -240,6 +244,10 @@ public class HudFragment{
 
             wavesMain.row();
 
+            if(Core.settings.getInt("AuxiliaryTable") == 2){
+                wavesMain.add(auxiliaryTools).left().top().row();
+            }
+
             addInfoTable(wavesMain.table().width(dsize * 5f + 4f).left().get());
 
             editorMain.name = "editor";
@@ -266,6 +274,11 @@ public class HudFragment{
             }).width(dsize * 5 + 4f);
             editorMain.visible(() -> shown && state.isEditor());
 
+            //map info/nextwave display
+            if(Core.settings.getInt("AuxiliaryTable") == 1){
+                cont.add(auxiliaryTools).left().top();
+            }
+
             //fps display
             cont.table(info -> {
                 info.name = "fps/ping";
